python/pyiceberg/expressions/base.py

```diff
@@ -169,16 +171,64 @@ def bind(self, schema: Schema, case_sensitive: bool) -> BoundReference[T]:
         return BoundReference(field=field, accessor=accessor)
 
 
-@dataclass(frozen=True)  # type: ignore[misc]
 class BoundPredicate(Bound[T], BooleanExpression):
-    term: BoundReference[T]
-    literals: Tuple[Literal[T], ...]
+    def __init__(self, term: BoundTerm[T], literals: tuple[Literal[T], ...] | Literal[T] | None = None):
```

Comment:
What do you think of just accumulating the positional arguments into a list of literals?
```suggestion
    def __init__(self, term: BoundTerm[T], *literals: Literal[T]):
```

Reply:
I like it! I may be misremembering. I thought I had gotten pushback in a similar situation using `*args` but let me have another go

Reply:
I think it would be nice for passing in the literal(s). We also use it in other places: https://github.com/apache/iceberg/blob/master/python/pyiceberg/schema.py#L63-L67 So I think we should be okay here.

Reply:
@Fokko Done!